examples/mutex.rs

```diff
@@ -107,19 +110,29 @@ mod example {
     }
 
     /// A guard holding a lock.
-    struct MutexGuard<'a, T>(Locked<'a, T>);
+    struct MutexGuard<'a, T> {
+        mutex: &'a Mutex<T>,
```

Comment:
Given that we only access `mutex.lock_ops`, wouldn't it be more appropriate to just place a pointer to `lock_ops` here?